### ISSUE TITLE
Monorepo fails to publish all packages when `--registry` contains trailing slash

### ISSUE THREAD

**Issue by 10hendersonm** [intuit/auto#1449]
relates to #1448 

**Describe the bug**

> ℹ️  I'm not super sure if this is a `lerna` bug, an `auto` bug, or an `artifactory` bug, but it was brought upon my team by the work implemented in #1412, which adds the `--registry` flag to the `lerna publish` command invoked within `publish` when calling `auto shipit`.

When calling `lerna publish` with the `--registry http://foo` command, adding a trailing slash to the registry (ex `lerna publish --registry http://foo/`) is causing shipit to fail as some (?) packages `401` against artifactory. This worked for us prior to `auto@9.48.3` as the `--registry` flag was not called and our `.npmrc` was respected instead.

I've worked around this in the meantime by just-in-time writing a non-trailing-slashed registry into the `publishConfig` of all packages' `package.json`.

**To Reproduce**

Run `auto shipit` with no `publishConfig.registry`, which will locate the registry from the root `.npmrc` but append a trailing slash to it (this is functionality from [registry-url](https://github.com/sindresorhus/registry-url/blob/master/index.js#L7)).

**Expected behavior**

I... I'm not really sure what's the best path here. A config option to disregard the `--registry` flag would improve my team's situation. I've heard that the trailing slash is necessary, but I'm also seeing it's breaking our deployments. Omitting `--registry` and setting our own `.npmrc` registry allows us to continue.

**Screenshots**

Note the `401` received in this output, but note that it follows several `201`s meaning those packages published successfully.

`npx auto shipit` output:
```bash
Command failed: npx auto shipit
ℹ info [DeleteAutoGeneratedVersionBranch] Applying Plugin
ℹ info [PushBranches] Applying Plugin
ℹ info [TagLatest] Applying Plugin
ℹ info [DryRunChangelog] Applying Plugin
⚠ warning lerna notice cli v3.13.1
lerna info ci enabled
lerna success found 6 packages
ℹ info Full Release notes for next release:
[CHANGELOG PREVIEW]
lerna notice cli v3.13.1
lerna info ci enabled
lerna info current version 21.0.0-canary.38760cb.0
lerna WARN force-publish all packages
lerna info Assuming all packages changed
lerna info auto-confirmed
lerna info execute Skipping git push
lerna info execute Skipping GitHub releases
lerna info publish Publishing packages to npm...
lerna notice Skipping all user and access validation due to third-party registry
lerna notice Make sure you're authenticated properly ¯\_(ツ)_/¯
lerna WARN ENOLICENSE Packages @my-scope/briefcase2, my-scope-but-not-scoped2, @my-scope/fedora2, @my-scope/glasses2, @my-scope/person2, and @my-scope/trenchcoat2 are missing a license.
lerna WARN ENOLICENSE One way to fix this is to add a LICENSE.md file to the root of this repository.
lerna WARN ENOLICENSE See https://choosealicense.com for additional guidance.
lerna success published @my-scope/trenchcoat2 21.0.0-beta.2
lerna notice
lerna notice package: @my-scope/trenchcoat2@21.0.0-beta.2
lerna notice === Tarball Contents ===
lerna notice 243B lib/trenchcoat.js
lerna notice 592B package.json
lerna notice 5.8kB CHANGELOG.md
lerna notice 137B README.md
lerna notice === Tarball Details ===
lerna notice name: @my-scope/trenchcoat2
lerna notice version: 21.0.0-beta.2
lerna notice filename: my-scope-trenchcoat2-21.0.0-beta.2.tgz
lerna notice package size: 1.6 kB
lerna notice unpacked size: 6.8 kB
lerna notice shasum: 884c9cd018f47eac84a7cf5171f03a81b5de72d5
lerna notice integrity: sha512-rtgjyA6ybFZCz[...]aba2GG5tO5FLA==
lerna notice total files: 4
lerna notice
lerna http fetch PUT 201 [artifactory]/npm/npm-my-scope/@my-scope%2ftrenchcoat2 241ms
lerna success published @my-scope/glasses2 21.0.0-beta.2
lerna notice
lerna notice package: @my-scope/glasses2@21.0.0-beta.2
lerna notice === Tarball Contents ===
lerna notice 280B lib/glasses.js
lerna notice 588B package.json
lerna notice 7.1kB CHANGELOG.md
lerna notice 132B README.md
lerna notice === Tarball Details ===
lerna notice name: @my-scope/glasses2
lerna notice version: 21.0.0-beta.2
lerna notice filename: my-scope-glasses2-21.0.0-beta.2.tgz
lerna notice package size: 1.7 kB
lerna notice unpacked size: 8.1 kB
lerna notice shasum: aff7e1c22bacf32341042df30664e3a837d723f6
lerna notice integrity: sha512-t//yEkt0hjbo/[...]mxq73Bj2d+yxQ==
lerna notice total files: 4
lerna notice
lerna http fetch PUT 201 [artifactory]/npm/npm-my-scope/@my-scope%2fglasses2 401ms
lerna success published @my-scope/briefcase2 21.0.0-beta.2
lerna notice
lerna notice package: @my-scope/briefcase2@21.0.0-beta.2
lerna notice === Tarball Contents ===
lerna notice 546B lib/briefcase.js
lerna notice 595B package.json
lerna notice 8.5kB CHANGELOG.md
lerna notice 194B README.md
lerna notice === Tarball Details ===
lerna notice name: @my-scope/briefcase2
lerna notice version: 21.0.0-beta.2
lerna notice filename: my-scope-briefcase2-21.0.0-beta.2.tgz
lerna notice package size: 2.0 kB
lerna notice unpacked size: 9.8 kB
lerna notice shasum: 06620ef9e9795043bcf4f1709c298bcc0c26ac37
lerna notice integrity: sha512-7wDt6YDhSUzSJ[...]PngyEfbp/DV2A==
lerna notice total files: 4
lerna notice
lerna http fetch PUT 201 [artifactory]/npm/npm-my-scope/@my-scope%2fbriefcase2 615ms
lerna http fetch PUT 401 [artifactory]/npm/npm-my-scope/@my-scope%2ffedora2 4112ms
lerna ERR! E401 Unable to authenticate, need: Basic realm="Artifactory Realm"
at ChildProcess.<anonymous> (/home/some-dir/my-scope-but-not-scoped2/node_modules/@auto-it/core/dist/utils/exec-promise.js:58:31)
at ChildProcess.emit (events.js:223:5)
at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)Error
at Object.execPromise (/home/some-dir/my-scope-but-not-scoped2/node_modules/@auto-it/core/dist/utils/exec-promise.js:15:22)
at /home/some-dir/my-scope-but-not-scoped2/node_modules/@auto-it/npm/dist/index.js:658:30
```

**Environment information:**

```txt
```

As with #1448, we're abstracting auto away into a separate CLI that makes this a little difficult (`auto` is not installed in our monorepo, `npx auto info` fails when provided with my Github Enterprise token as opposed to a public github token, etc). I can provide the output if it's needed.

**Additional context**

I'm not 100% sure work output is required from here, but I did want to type this up as supporting evidence towards #1448.

**Comment by hipstersmoothie** [comment]
Sorry about that! This doesn't seem like a fun problem! I'm trying to reproduce on my end at the moment.

**Comment by hipstersmoothie** [comment]
Oddly both trailing and non-trailing work on publish npm

**Comment by hipstersmoothie** [comment]
I'm really thinking that this is an artifactory problem. They have had lots of weirdness in the past. I'd rather not have to add another config option but if It's necessary we can.

**Comment by hipstersmoothie** [comment]
Got artifactory creds today. will try to reproduce tomorrow

**Comment by 10hendersonm** [comment]
Ah totally forgot to swing back by my laptop when I saw the update..

Honestly, with #1448 handled I'm much less concerned about this. I mostly typed it up as justification for that one. As stated, we've worked around this internally (by just-in-time injecting `publishConfig.registry`), and honestly it's not an issue with your work, it's an issue upstream in `lerna` or `artifactory`.

I'm OK with this being closed, and I'm sorry if you spent a lot of effort getting Artifactory working.

**Comment by hipstersmoothie** [comment]
So I was able to recreate this locally but I could only get that error to come up when I was doing my auth wrong. From what I've read artifactory only supports legacy auth for npm. 

This did not work

```
//registry.MINE.org/:_authToken=MY_TOKEN
```

but this did work

```
registry=https://registry.MINE.com
_auth = TOKEN
always-auth = true
email = Andrew_Lisowski@company.com
```

Which I created by running

```
cp .npmrc .npmrc.bak 2>/dev/null
echo "registry=https://registry.npmjs.intuit.com" > .npmrc
curl -u MY_EMAIL:MY_ARTIFACTORY_TOKEN https://artifact.MINE.com/artifactory/api/npm/auth >> .npmrc
```

But this might be a different issue. I had no issues with the trailing slash.

**Comment by 10hendersonm** [comment]
We're currently publishing with `_auth = TOKEN`. I do have one commit where I'd tried [enabling `legacyAuth` on the NPM plugin](https://intuit.github.io/auto/docs/generated/npm#legacyauth). We generate the token directly off of artifactory, and when I deconvert it from base64 to ASCII it doesn't turn into `username:password` format, so I'm unsure what exactly it is. So, the commit where I had tried `legacyAuth` likely wasn't using `username:password`.

I've got it running and there's a little extra commit noise, I'm not broken up over it. Whatever the issue is, it's much deeper than the `auto` layer, so I'm willing to close.

Big thanks for chucking #1450 in though!
